### DISCUSSION TITLE
Sound and text improvements

### DISCUSSION THREAD

**Discussion by jtothebell** [jtothebell/fake-08#192] [General]
All Platforms

- Add support for new `stat` values (46-56) for audio (thanks @fringd )
- Rewrite text rendering to support custom fonts and most remaining P8SCII control codes (audio control codes are parsed, but not played) and speed up text rendering (probably only noticeable on slow platforms like Bittboy and 3ds)
- Add a corssfade between notes to more closely match PICO-8 behavior and fix some audio clipping (thanks @fringd )
- Add custom instrument support (thanks @fringd )
- add `inext` function to z8lua

Libretro platforms

- Add support for analog input to emulate the mouse (thanks for the suggestion and start of implementation @medeirosT)
- Add support for loading carts from memory in addition to the file system (may help with loading carts on Android devices with scoped storage)
- 

<hr /><em>This discussion was created from the release <a href='https://github.com/jtothebell/fake-08/releases/tag/v0.0.2.20'>Sound and text improvements</a>.</em>

**Comment by Airwalk16** [comment]
jon Bell my man, you do know this release doesn't work for shit, right? At least the android ones. Renaming the .info file to fake08_libretro.info, dopping it into the cores file, then installing it, using the .so file doesn't even recognize p8 file extensions. It literally says no core available to play content once loaded. Trying to load content straight from load content option, p8 carts literally don't even show up. Your last release ”initial build" is the only working android and even it still has some issues with a number of carts. Mainly newer ones from lexaloffle BBS. Lol
